src/module.c

```diff
@@ -9546,18 +9546,24 @@ void modulesCron(void) {
      * that we didn't use for the last cron period. */
 
     /* Limit the max client count to be freed at once to avoid latency spikes.*/
-    int iteration = 500;
-    /* We are freeing clients if we have more than 16 unused clients. Keeping
+    int iteration = 50;
+    /* We are freeing clients if we have more than 8 unused clients. Keeping
      * small amount of clients to avoid client allocation costs if temporary
      * clients are required after some idle period. */
-    const int min_client = 16;
+    const int min_client = 8;
     while (iteration > 0 && moduleTempClientCount > 0 && moduleTempClientMinCount > min_client) {
         client *c = moduleTempClients[--moduleTempClientCount];
         freeClient(c);
         iteration--;
         moduleTempClientMinCount--;
     }
     moduleTempClientMinCount = moduleTempClientCount;
+
+    /* Shrink moduleTempClients array itself if it is wasting some space */
+    if (moduleTempClientCap > 32 && moduleTempClientCap > moduleTempClientCount * 2) {
```

Comment:
i think this can have a resonance effect.
you multiply by 2, and then shrink to half.
maybe the shrinking should be only when there's 4 times waste?

Reply:
Done.